packages/frontend/app/(dashboard)/course/[cid]/insights/utils/types.ts

```diff
@@ -0,0 +1,285 @@
+  [key: string]: ChartComponent
+} = {
+  QuestionTypeBreakdown: {
+    chartType: 'Pie',
+    categoryKeys: true,
+    props: {
+      includeLegend: true,
+      includeTooltip: true,
+      innerRadius: 0,
+      showLabels: true,
+      size: 'lg',
+      labelFormatter: (label: string) => label,
+    },
+  },
+  AverageTimesByWeekDay: {
+    chartType: 'Bar',
+    props: {
+      includeLegend: true,
+      includeTooltip: true,
+      size: 'lg',
+      valueFormatter: (label: any) => label + ' minutes',
+    },
+  },
+  HelpSeekingOverTime: {
+    chartType: 'Area',
+    props: {
+      includeLegend: true,
+      includeTooltip: true,
+      size: '4xl',
+      aspectRatio: 3,
+      tickFormatter: (label) =>
+        new Date(label).toLocaleString('en-US', {
+          year: '2-digit',
+          month: 'short',
+          day: 'numeric',
+        }),
+      labelFormatter: (label, payload) => {
+        if (payload != undefined) {
+          return new Date((payload as any[])[0].payload.key).toLocaleString(
+            'en-US',
+            {
+              year: 'numeric',
+              month: 'short',
+              day: 'numeric',
+            },
+          )
+        } else {
+          return label
+        }
+      },
+    },
+  },
+  HumanVsChatbot: {
+    chartType: 'Bar',
+    props: {
+      includeLegend: true,
+      includeTooltip: true,
+      size: 'lg',
+      valueFormatter: (label: any) =>
+        parseInt(label) > 1 || parseInt(label) == 0
+          ? label + ' answers'
+          : label + ' answer',
+      tickFormatter: (label) => label,
+    },
+  },
+  HumanVsChatbotVotes: {
+    chartType: 'Bar',
+    props: {
+      includeLegend: true,
+      includeTooltip: true,
+      size: 'lg',
+      valueFormatter: (label: any) =>
+        parseInt(label) > 1 || parseInt(label) == 0
+          ? label + ' votes'
+          : label + ' vote',
+      tickFormatter: (label) => label,
+    },
+  },
+}
+
+export const gantt_charts: {
+  [key: string]: GanttChartComponent
+} = {
+  MostActiveTimes: {
+    props: {
+      includeTooltip: true,
+      size: '4xl',
+      labelFormatter: (
+        label: string,
+        payload: TooltipPayload<string, NameType>[],
+      ) => {
+        if (payload != undefined) {
+          const p = (payload as any[])[0]?.payload
+          return (
+            numToWeekday(p.Weekday) + `  (${getMinutesToTime(p.key as number)})`
+          )
+        } else {
+          return label
+        }
+      },
+      xTickFormatter: (label: string) => {
+        const num = parseInt(label)
+        if (isNaN(num)) {
+          return ''
+        }
+        return getMinutesToTime(num)
+      },
+      yTickFormatter: (label: string) =>
+        numToWeekday(parseInt(label)).substring(0, 3),
+      valueFormatter: (label: any, key: any, body: any) => {
+        if (key == 'key') {
+          const payload: any = body.payload
+          return 'Questions: ' + payload?.Amount ?? 0
```

Comment:
My vscode caught an error here `Right operand of ?? is unreachable because the left operand is never nullish.`

```suggestion
          return 'Questions: ' + (payload?.Amount ?? 0)
```
I'm assuming it was meant to be like this

Reply:
thanks, addressed